docs/data/data-grid/column-dimensions/ColumnAutosizingHeaderFilters.js

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import Rating from '@mui/material/Rating';
+import { DataGridPremium } from '@mui/x-data-grid-premium';
+import { randomRating, randomTraderName } from '@mui/x-data-grid-generator';
+
+function renderRating(params) {
+  return <Rating readOnly value={params.value} />;
+}
+
+function useData(length) {
+  return React.useMemo(() => {
+    const names = [
+      'Nike',
+      'Adidas',
+      'Puma',
+      'Reebok',
+      'Fila',
+      'Lululemon Athletica Clothing',
+      'Varley',
+    ];
+
+    const rows = Array.from({ length }).map((_, id) => ({
+      id,
+      brand: names[id % names.length],
+      rep: randomTraderName(),
+      rating: randomRating(),
+    }));
+
+    const columns = [
+      { field: 'id', headerName: 'Brand ID' },
+      { field: 'brand', headerName: 'Brand name' },
+      { field: 'rep', headerName: 'Representative' },
+      {
+        field: 'rating',
+        headerName: 'Rating',
+        renderCell: renderRating,
+        display: 'flex',
+      },
+    ];
```

Comment:
Nit: It's always better to define columns outside of the component/hook for referential stability